completion_aggregator/api/v1/views.py

```diff
@@ -388,3 +393,136 @@ def get(self, request, course_key):
+            later.
+
+        exclude_roles (optional):
+            A comma separated list of roles to exclude from the results.
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid value was sent for requested_fields and cohorts,
+        * 404 If the course is not cohorted
+
+        Example response:
+
+            {
+                "course_key": "edX/toy/2012_Fall",
+                "filters": {
+                    "cohorts": 2,
+                    "exclude_roles": [
+                        "beta",
+                        "staff"
+                    ]
+                },
+                "results": [
+                    {
+                        "mean_completion": {
+                            "earned": 3.4,
+                            "possible": 8.0,
+                            "percent": 0.425
+                        }
+                    }
+                ]
+            }
+
+    """
+    course_completion_serializer = serializers.CourseCompletionStatsSerializer
+    block_completion_serializer = serializers.BlockCompletionSerializer
+
+    def _parse_cohort_filter(self, cohort_filter):
+        """
+        Helper function to parse cohort filter query parameter.
+        """
+        if cohort_filter is not None:
+            try:
+                cohort_filter = int(cohort_filter)
+            except TypeError:
+                raise ParseError(
+                    'could not parse cohort_filter={!r} as an integer'.format(
+                        cohort_filter,
+                    )
+                )
+        return cohort_filter
+
+    def get(self, request, course_key):
+        """
+        Handler for GET requests
+        """
+        course_key = CourseKey.from_string(course_key)
+        paginator = self.pagination_class()  # pylint: disable=not-callable
+        requested_fields = self.get_requested_fields()
+        roles_to_exclude = self.request.query_params.get('exclude_roles', '').split(',')
+        cohort_filter = self._parse_cohort_filter(
+            self.request.query_params.get('cohorts'))
+
+        enrollments = UserEnrollments().get_course_enrollments(course_key)
+        paginated = paginator.paginate_queryset(
+            enrollments, self.request, view=self)
+        aggregator_qs = self.get_queryset().filter(
+            course_key=course_key,
+            user__in=[enrollment.user for enrollment in paginated])
+        if roles_to_exclude:
+            aggregator_qs = aggregator_qs.exclude(
+                user__courseaccessrole__role__in=roles_to_exclude)
+        if cohort_filter is not None:
+            aggregator_qs = aggregator_qs.exclude(
+                user__cohortmembership__course_user_group__pk=cohort_filter)
+        completions = aggregator_qs.aggregate(
+            possible=Avg('possible'),
+            earned=Avg('earned'),
+            percent=Sum('earned') / Sum('possible'))
```

Comment:
Nit, rename this `completion_stats`.  `completions` sounds like a collection of Aggregator or BlockCompletion objects, but since we've aggregated (in the database sense) we no longer have a collection.